src/ngraph/op/experimental/batch_dot.hpp

```diff
@@ -0,0 +1,62 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        /// \brief Dot product for a batch of Rank 2 tensors.
+        /// The inputs are expected to be Rank 3, where the first dim is the
+        /// batch size and must be the same for both inputs. The last two dims
+        /// are the shape of matrices, i.e. `(batch_size, :, :)`.
+        /// For example, for `a` with shape `(batch_size, n, k)`, and `b` with
+        /// shape `(batch_size, k, m)`, the result of BatchDot will have shape
+        /// `(batch_size, n, m)`, and `BatchDot(a, b)[i] = Dot(a[i], b[i])`.
+        class BatchDot : public Op
+        {
+        public:
+            /// \brief Constructs a batch dot product operation.
+            ///
+            /// \param arg0 The node producing the first argument.
+            /// \param arg1 The node producing the second argument.
+            /// \param transpose_0 Apply transpose to arg0.
+            /// \param transpose_1 Apply transpose to arg1.
```

Comment:
We have ops for doing these transposes, so let's use them instead of complicating this op. Backends can fuse them if they want to.

Reply:
Sounds good, I'll leave make the core op simple without transpose and leave the CPU version of batch dot and fusion unchanged in this PR. Additional fusion can be done later, but don't want to block PDPD to start using batchmatmul.

Reply:
Fixed.